core/src/main/scala/com/amadeus/dataio/config/fields/StreamingTriggerConfigurator.scala

```diff
@@ -0,0 +1,36 @@
+package com.amadeus.dataio.config.fields
+
+import com.typesafe.config.Config
+import org.apache.spark.sql.streaming.Trigger
+
+import scala.concurrent.duration.Duration
+import scala.util.Try
+
+/**
+ * Retrieve the trigger to be used from the configuration.
+ *
+ * See documentation: [[https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#continuous-processing]]
+ *
+ * Trigger  = "" AvailableNow or Continuous. Optional in case of ProcessingTime trigger.
+ * Duration = "1 minute". Optional in case of AvailableNow trigger.
+ */
+trait StreamingTriggerConfigurator {
+
+  /**
+   * @param config The typesafe Config object holding the configuration.
+   * @return the trigger of None is not defined.
+   * @throws IllegalArgumentException in case the combination of Trigger and Duration is not supported.
+   */
+  def getStreamingTrigger(implicit config: Config): Option[Trigger] = {
```

Comment:
For the sake of consistency, I would remove the Option type here. Other fields do not return Option, leaving the choice of making it optional to the calling function.

Changing this can be discussed, but we'll need to update the other fields as well.

Reply:
To me it would push the robustness checks to the caller function and add code duplication.

Reply:
I agree, it was done this way to mimic the behavior of TypeSafe config getter functions due to it being written in Java. But this is more idiomatic for Scala so let's keep it as you suggested.

I'll keep trace that we need to update the other configurators to follow a similar pattern.